wolfssl/wolfcrypt/wc_port.h

```diff
@@ -371,6 +430,54 @@
 
     #endif /* USE_WOLFSSL_LINUXKM_PIE_REDIRECT_TABLE */
 
+#ifdef WOLFSSL_LINUXKM_SIMD
+
+#ifdef WOLFSSL_LINUXKM_SIMD_X86
+
+#ifdef WOLFSSL_LINUXKM_SIMD_X86_IRQ_ALLOWED
+    extern __must_check int allocate_wolfcrypt_irq_fpu_states(void);
+    extern void free_wolfcrypt_irq_fpu_states(void);
+    extern __must_check int save_vector_registers_x86(void);
+    extern void restore_vector_registers_x86(void);
+#else /* !WOLFSSL_LINUXKM_SIMD_X86_IRQ_ALLOWED */
+    static __must_check inline int save_vector_registers_x86(void) {
```

Comment:
Is it possible to move these three static inline function into memory.c next to `save_vector_registers_x86`?

Reply:
best to leave these as inlines for efficiency (the calls they use are actually inlines), but I had to turn them into macros, to let them use wolfcrypt error codes.

alas I forgot to similarly refactor the ARM versions of them... but that's placeholder code anyway, untested and only hit by the arm-asm use case, which I haven't made a start on yet.